cmd/containerboot/tailscaled.go

```diff
@@ -166,3 +169,64 @@ func tailscaleSet(ctx context.Context, cfg *settings) error {
 	}
 	return nil
 }
+
+func watchTailscaledConfigChanges(ctx context.Context, path string, lc *tailscale.LocalClient, errCh chan error) {
+	var (
+		tickChan          <-chan time.Time
+		tailscaledCfgDir  = filepath.Dir(path)
+		prevTailscaledCfg []byte
+	)
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Printf("tailscaled config watch: failed to create fsnotify watcher, timer-only mode: %v", err)
+		ticker := time.NewTicker(5 * time.Second)
+		defer ticker.Stop()
+		tickChan = ticker.C
+	} else {
+		defer w.Close()
+		if err := w.Add(tailscaledCfgDir); err != nil {
+			errCh <- fmt.Errorf("failed to add fsnotify watch: %w", err)
+			return
+		}
+	}
+	// kubelet mounts Secrets to Pods using a series of symlinks, one of
+	// which is <mount-dir>/..data that Kubernetes recommends consumers to
+	// use if they need to monitor changes
+	// https://github.com/kubernetes/kubernetes/blob/v1.28.1/pkg/volume/util/atomic_writer.go#L39-L61
+	const kubeletMountedCfg = "..data"
+	toWatch := filepath.Join(tailscaledCfgDir, kubeletMountedCfg)
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case err := <-w.Errors:
+			errCh <- fmt.Errorf("watcher error: %w", err)
+			return
+		case <-tickChan:
```

Comment:
This case is empty and Go doesn't support `fallthrough` for `select` statements (although I suppose we wouldn't want the code to be exactly the same, as there's no `event` for the timer case). Maybe make a function out of it, that returns something like `(updated bool, err error)`?

Reply:
Should be fixed, I cannot remember writing this anymore, but all the code starting from this line was meant to go after select on interesting events / timer ticks https://github.com/tailscale/tailscale/pull/14342/commits/6ced39ff0f7006f3caa2576bd6577f25a5e0ce81#diff-e77931915d11ab8f75c283d2bf335012fcc0b61ed441106c3dd7b59279aa72bbR217

Reply:
Nice, that's really elegant 👍